src/qibo/engines/__init__.py

```diff
@@ -1,6 +1,51 @@
+from qibo import config
+from qibo.config import log, raise_error
 from qibo.engines.numpy import NumpyEngine
+from qibo.engines.tensorflow import TensorflowEngine
 
-engine = NumpyEngine()
 
-# TODO: Implement engine setter, similar to ``qibo.set_backend()``
-# This global engine will be used as default by ``circuit.execute()
\ No newline at end of file
+class GlobalSimulator:
+    """The global simulator will be used as default by ``circuit.execute()``."""
+    # TODO: Implement precision setters
+    # TODO: Implement device setters
+
+    _instance = NumpyEngine()
+
+    def __new__(cls):
+        """Creates singleton instance."""
+        return cls._instance
+
+    @classmethod
+    def set_backend(cls, backend, platform):
+        if not config.ALLOW_SWITCHERS and backend != cls._instance.name:
+            log.warning("Backend should not be changed after allocating gates.")
+        # TODO: Use ``profiles.yml`` here
```

Comment:
Not sure about keeping this `profiles.yml`. Seems to me that a simple env variable should be more than sufficient.